browser-switch/src/main/java/com/braintreepayments/browserswitch/BrowserSwitchClient.java

```diff
@@ -0,0 +1,332 @@
+            BrowserSwitchPersistentStore.getInstance(), returnUrlScheme
+        );
+    }
+
+    @VisibleForTesting
+    static BrowserSwitchClient newInstance(
+        BrowserSwitchConfig config, ActivityFinder activityFinder,
+        BrowserSwitchPersistentStore persistentStore, String returnUrlScheme
+    ) {
+        return new BrowserSwitchClient(config, activityFinder, persistentStore, returnUrlScheme);
+    }
+
+    private BrowserSwitchClient(
+            BrowserSwitchConfig config, ActivityFinder activityFinder,
+            BrowserSwitchPersistentStore persistentStore, String returnUrlScheme
+    ) {
+        this.config = config;
+        this.activityFinder = activityFinder;
+        this.persistentStore = persistentStore;
+        this.returnUrlScheme = returnUrlScheme;
+    }
+
+    /**
+     * Open a browser or <a href="https://developer.chrome.com/multidevice/android/customtabs">Chrome Custom Tab</a>
+     * with the given url from an Android fragment. The fragment must be attached to activity when invoking this method.
+     *
+     * @param requestCode the request code used to differentiate requests from one another.
+     * @param uri the url to open.
+     * @param fragment the fragment used to start browser switch
+     */
+    void start(int requestCode, Uri uri, Fragment fragment) {
+        if (fragment instanceof BrowserSwitchListener) {
+            start(requestCode, uri, fragment, (BrowserSwitchListener) fragment);
+        } else {
+            throw new IllegalArgumentException("Fragment must implement BrowserSwitchListener.");
+        }
+    }
+
+    /**
+     * Open a browser or <a href="https://developer.chrome.com/multidevice/android/customtabs">Chrome Custom Tab</a>
+     * with the given url from an Android fragment. The fragment must be attached to activity when invoking this method.
+     *
+     * @param requestCode the request code used to differentiate requests from one another.
+     * @param uri the url to open.
+     * @param fragment the fragment used to start browser switch
+     * @param listener the listener that will receive browser switch callbacks
+     */
+    void start(int requestCode, Uri uri, Fragment fragment, BrowserSwitchListener listener) {
+        FragmentActivity activity = fragment.getActivity();
+        if (activity != null) {
+            start(requestCode, uri, activity, listener);
+        } else {
+            throw new IllegalStateException("Fragment must be attached to an activity.");
+        }
+    }
+
+    /**
+     * Open a browser or <a href="https://developer.chrome.com/multidevice/android/customtabs">Chrome Custom Tab</a>
+     * with the given url from an Android activity.
+     *
+     * @param requestCode the request code used to differentiate requests from one another.
+     * @param uri the url to open.
+     * @param activity the activity used to start browser switch
+     */
+    @SuppressWarnings("SameParameterValue")
```

Comment:
Why was this needed?

Reply:
Android Studio flagged a warning because the demo project only calls this method once. The IDE thinks the parameter `requestCode` could potentially be a constant. I suppressed the warning to make the code lint-free, but we may be able to ignore this linting error with a lint config in the future.